table/server/common/src/main/java/alluxio/table/common/udb/UdbMountSpec.java

```diff
@@ -0,0 +1,589 @@
+      if (someSet == null) {
+        return elements;
+      } else {
+        someSet.addAll(elements);
+        return someSet;
+      }
+    }
+
+    private static <T> Set<T> replaceNullWithEmpty(Set<T> someSet) {
+      if (someSet == null) {
+        return Collections.emptySet();
+      }
+      return someSet;
+    }
+  }
+
+  /**
+   * Builder for include/exclude list.
+   * @param <I> the type contained in the included list
+   * @param <IBUILDERT> builder type for {@link I}
+   * @param <E> the type contained in the excluded list
+   * @param <EBUILDERT> builder type for {@link E}
+   * @param <OUT> the output type
+   */
+  abstract static class IncludeExcludeBuilder<I, IBUILDERT extends BaseBuilder<I, IBUILDERT>,
+                                              E, EBUILDERT extends BaseBuilder<E, EBUILDERT>,
+                                              OUT> {
+    private final IBUILDERT mInclude;
+    private final EBUILDERT mExclude;
+
+    protected IncludeExcludeBuilder(IBUILDERT includeBuilder, EBUILDERT excludeBuilder) {
+      mInclude = includeBuilder;
+      mExclude = excludeBuilder;
+    }
+
+    /**
+     * Returns the included entry's builder.
+     */
+    public IBUILDERT include() {
+      return mInclude;
+    }
+
+    /**
+     * Returns the excluded entry's builder.
+     */
+    public EBUILDERT exclude() {
+      return mExclude;
+    }
+
+    /**
+     * Builds the type.
+     */
+    abstract OUT build();
+  }
+
+  /**
+   * Builder for simple names and patterns.
+   */
+  public static final class SimpleNamePatternWrapperBuilder
+      extends BaseBuilder<SimpleNamePatternWrapper, SimpleNamePatternWrapperBuilder> {
+    @Override
+    SimpleNamePatternWrapperBuilder self() {
+      return this;
+    }
+
+    @Override
+    SimpleNamePatternWrapper build() {
+      mNames = BaseBuilder.replaceNullWithEmpty(mNames);
+      mPatterns = BaseBuilder.replaceNullWithEmpty(mPatterns);
+      SimpleNamePatternWrapper built = new SimpleNamePatternWrapper(mNames, mPatterns);
+      // release references
+      mNames = null;
+      mPatterns = null;
+      return built;
+    }
+  }
+
+  /**
+   * Builder for table with partition specification.
+   */
+  public static final class TablePartitionNamePatternWrapperBuilder
+      extends BaseBuilder<TablePartitionNamePatternWrapper,
+      TablePartitionNamePatternWrapperBuilder> {
+    private Map<String, Set<SimpleWrapper>> mTablePartMap;
+
+    @Override
+    TablePartitionNamePatternWrapperBuilder self() {
+      return this;
+    }
+
+    /**
+     * Attaches partition specification to a table.
+     * @param tableName the table name
+     * @param builder builder for partition specification
+     * @return the builder
+     */
+    public TablePartitionNamePatternWrapperBuilder addPartition(String tableName,
+                                                                SimpleWrapperBuilder builder) {
+      return addPartition(tableName, builder.build());
```

Comment:
Here the second parameter is the builder instead of the built `SimpleWrapper`, since `SimpleWrapper` is package-private, and therefore `SimpleWrapperBuilder.build()` is made package-private. I tried to hide these inner classes that are implementation details from outside code.